src/selectors/per-thread/thread.js

```diff
@@ -385,6 +385,10 @@ export function getThreadSelectorsPerThread(
     ProfileSelectors.getProfileViewOptions(state).perThread[threadsKey] ||
     defaultThreadViewOptions;
 
+  /** Check to see if there are any timing samples for this thread. */
+  const getHasSamples: Selector<boolean> = (state) =>
+    Boolean(getThread(state).samples.length);
```

Comment:
> We may even go the extra mile and doing the full algorithm that you reused above (check if the only used stack is "(root)"). What do you think?

1. Should the "Data Source" options also use this to decide whether or not they should be visible?
2. I still need the `thread` or at least the tables inside the thread for the "(root)" check, so should these still be in some per-thread file? Or should I just take thread as an argument while keeping the function in profile-data.js?

Reply:
1. Yeah, once the code is extracted to profile-data, both locations can use this shared function
2. and yes, the extracted function could take the full thread as a second parameter for example. I'm suggesting here to use both the table as first argument and the full thread as second argument, so that you can pass different tables as the first argument. But please tell me if that sounds crazy

Reply:
I think I've done what you said, but I'll leave you to resolve this conversation in case I misunderstood something.